editor/src/components/editor/loading-screen.tsx

```diff
@@ -0,0 +1,174 @@
+    [importState],
+  )
+
+  const projectId = useEditorState(
+    Substores.restOfEditor,
+    (store) => store.editor.id,
+    'LoadingEditorComponent projectId',
+  )
+
+  const cleared = React.useRef(false)
+
+  const currentOperationToShow: {
+    text: React.ReactNode
+    id: string
+    timeDone: number | null | undefined
+    timeStarted: number | null | undefined
+  } | null = React.useMemo(() => {
+    if (totalImportResult.importStatus.status == 'not-started') {
+      if (projectId == null) {
+        return {
+          text: 'Loading Editor...',
+          id: 'loading-editor',
+          timeDone: null,
+          timeStarted: null,
+        }
+      } else {
+        return {
+          text: `Parsing files`,
+          id: 'parseFiles',
+          timeDone: null,
+          timeStarted: null,
+        }
+      }
+    }
+    for (const op of importState.importOperations) {
+      if (op?.children?.length == 0 || op.type == 'refreshDependencies') {
+        if (op.timeStarted != null && op.timeDone == null) {
+          return {
+            text: getImportOperationTextAsJsx(op),
+            id: op.id ?? op.type,
+            timeDone: op.timeDone,
+            timeStarted: op.timeStarted,
+          }
+        }
+      }
+      if (op.type !== 'refreshDependencies') {
+        for (const child of op.children ?? []) {
+          if (child.timeStarted != null && child.timeDone == null) {
+            return {
+              text: getImportOperationTextAsJsx(child),
+              id: child.id ?? child.type,
+              timeDone: child.timeDone,
+              timeStarted: child.timeStarted,
+            }
+          }
+        }
+      }
+    }
+    return {
+      text: 'Loading Editor...',
+      id: 'loading-editor',
+      timeDone: null,
+      timeStarted: null,
+    }
+  }, [totalImportResult, importState.importOperations, projectId])
+
+  const shouldBeCleared = React.useMemo(() => {
+    return (
+      cleared.current ||
+      (totalImportResult.importStatus.status == 'done' &&
+        (githubRepo == null || totalImportResult.result == 'criticalError')) ||
+      totalImportResult.importStatus.status == 'paused'
+    )
+  }, [totalImportResult, githubRepo])
+
+  React.useEffect(() => {
+    if (shouldBeCleared) {
+      const loadingScreenWrapper = document.getElementById('loading-screen-wrapper')
+      if (loadingScreenWrapper != null) {
+        loadingScreenWrapper.remove()
+      }
+    }
+  }, [shouldBeCleared])
+
+  const portal = React.useRef(document.getElementById('loading-screen-progress-bar-portal')).current
+  const hasMounted = React.useRef(false)
+  if (portal == null) {
+    return null
+  }
+
+  if (shouldBeCleared) {
+    cleared.current = true
+    return null
+  }
+
+  if (!hasMounted.current) {
+    portal.innerHTML = ''
+    hasMounted.current = true
+  }
+
+  const themeStyle =
+    currentTheme === 'dark'
+      ? `
+    .editor-loading-screen { background-color: ${colorTheme.bg6.value} }
```

Comment:
in a sense this is "patching", since we want to match the CSS behavior in [index.html](https://github.com/concrete-utopia/utopia/blob/a1c03084719bdfff4757fb8fc3d2950817941f87/editor/src/templates/index.html#L94-L112) where the dark mode "overrides" the light mode defaults (that also apply here).
I didn't want to change anything in case we have a light theme, so to not create a possible shift in color from the light "default" that is defined in the base `index.html` css.